### PR TITLE
Vault-11623: OSS changes for seal config and env vars

### PR DESCRIPTION
**Pull request by rculpepper**
This is the OSS side of the changes for seal config checks and using env vars for seals

### CODE REVIEW

File: command/server/config.go [hunk 1]
```diff
@@ -494,13 +494,17 @@ func CheckConfig(c *Config, e error) (*Config, error) {
...
-			return nil, errors.New("seals: two seals provided but both are disabled")
-		case !c.Seals[0].Disabled && !c.Seals[1].Disabled:
-			return nil, errors.New("seals: two seals provided but neither is disabled")
+	if err := c.checkSealConfig(); err != nil {
+		return nil, err
+	}
+
+	sealMap := make(map[string]*configutil.KMS)
+	for _, seal := range c.Seals {
+		if _, ok := sealMap[seal.Name]; ok {
```
Comment (sgmiller) [inline command/server/config.go:503]: Honest question, could go either way, should we be case sensitive here or more restrictive?

Comment (sgmiller) [reply command/server/config.go:503]: Also, if the seal is not named, will two unnamed seals conflict?

File: internalshared/configutil/kms.go [hunk 2]
```diff
@@ -102,6 +107,28 @@ func parseKMS(result *[]*KMS, list *ast.ObjectList, blockName string, maxKMS int
...
+			}
+			delete(m, "priority")
+
+			if priority < 1 {
+				return multierror.Prefix(fmt.Errorf("invalid priority in kms type %q: %d", key, priority), fmt.Sprintf("%s.%s", blockName, key))
+			}
+		}
+
+		name := strings.ToLower(key)
+		if v, ok := m["name"]; ok {
```
Comment (sgmiller) [inline internalshared/configutil/kms.go:124]: Oh, or per above are we tolowering here so it doesn't matter.

File: internalshared/configutil/kms.go [hunk 3]
```diff
@@ -328,7 +362,7 @@ func GetOCIKMSKMSFunc(kms *KMS, opts ...wrapping.Option) (wrapping.Wrapper, map[
 
 var GetTransitKMSFunc = func(kms *KMS, opts ...wrapping.Option) (wrapping.Wrapper, map[string]string, error) {
 	wrapper := transit.NewWrapper()
-	wrapperInfo, err := wrapper.SetConfig(context.Background(), append(opts, wrapping.WithConfigMap(kms.Config))...)
+	wrapperInfo, err := wrapper.SetConfig(context.Background(), append(opts, wrapping.WithDisallowEnvVars(true), wrapping.WithConfigMap(kms.Config))...)
```
Comment (sgmiller) [inline internalshared/configutil/kms.go:365]: Nice!

File: internalshared/configutil/kms.go [hunk 4]
```diff
@@ -350,3 +384,34 @@ var GetTransitKMSFunc = func(kms *KMS, opts ...wrapping.Option) (wrapping.Wrappe
 func createSecureRandomReader(conf *SharedConfig, wrapper wrapping.Wrapper) (io.Reader, error) {
 	return rand.Reader, nil
 }
+
+func getEnvConfig(kms *KMS) map[string]string {
```
Comment (sgmiller) [inline internalshared/configutil/kms.go:388]: Really like this enhancement

File: command/server/config.go (hunk 1 above)
Comment (rculpepper) [reply command/server/config.go:503]: Two unnamed seals that have the same type will conflict, since they'll both be given the name of the seal type

**Review by sgmiller — APPROVED** [review]

**Pull request merged**